cli/main.c

```diff
@@ -157,33 +145,67 @@ int main(int argc, char** argv) {
   // FIXME: this is temp till arg parse implemented.
   PkResult result;
 
-  if (argc == 1) {
+  static const char* const usage[] = {
+    "pocket [option] ... [-c cmd | file | -] [arg] ...",
+    NULL,
+  };
+
+  bool debug = false, quiet = false, version = false, help = false;
+  const char** cmd = NULL;
+
+  struct argparse_option cli_opts[] = {
+    OPT_GROUP("Basic options"),
+    OPT_STRING('c', "cmd", &cmd, "Evaluate and run the passed string."),
+    OPT_BOOLEAN('d', "debug", &debug, "Compile and run a debug version."),
+    OPT_BOOLEAN('h', "help", &help, "Prints this help message and exit."),
+    OPT_BOOLEAN('q', "quiet", &quiet,
+               "Don't print version and copyright statement on REPL startup."),
+    OPT_BOOLEAN('v', "version", &version,
+               "Prints the pocketlang version number and exit."),
+    OPT_END(),
+  };
+
+  struct argparse argparse;
+  argparse_init(&argparse, cli_opts, usage, 0);
+  argparse_describe(&argparse,
+        "\nPocketlang is a small and fast functional language written in C.",
+        "\nCopyright(c) 2020 - 2021 ThakeeNathees." \
+        "\nFree and open source software under the terms of the MIT license.");
+  argc = argparse_parse(&argparse, argc, argv);
+
+  if (debug) {
+    options.debug = debug;
     options.repl_mode = true;
     repl(vm, &options);
-
-  } if (argc >= 3 && strcmp(argv[1], "-c") == 0) {
-
+  } else if (help) {
+      argparse_usage(&argparse);
+      return 0;
+  } else if (version)
+    printf("pocketlang %s\n", PK_VERSION_STRING);
+  else if (strcmp(argv[1], "-c") == 0) {
```

Comment:
You've already parsed it with the `argparse` library. Use the `cmd` option here.